packages/pages/src/dev/server/middleware/sendAppHTML.ts

```diff
@@ -20,7 +23,8 @@ export default async function sendAppHTML(
   templateModuleInternal: TemplateModuleInternal<any, any>,
   props: TemplateRenderProps,
   vite: ViteDevServer,
-  pathname: string
+  pathname: string,
+  projectStructure: ProjectStructure
 ) {
   if (templateModuleInternal.render) {
```

Comment:
so client/server templates are only supported with a default export? I think that makes sense but just want to double check.

Reply:
No, it's a named export of `render` in client/server, not default. The line you commented on is the older custom render func in the **template** if you want to go full custom.